### DISCUSSION TITLE
Iconify 图标无法显示

### DISCUSSION THREAD

**Discussion by FYWinds** [vuepress-theme-hope/vuepress-theme-hope#3125] [Q&A] [answered]
## 问题表现
Iconify 图标无法显示
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/74457127/508a29f4-38f0-4cd7-86b9-5d9054a2f25b)
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/74457127/d584c50f-98ad-492f-a1f5-ca1f24262829)

尝试过文档中所写的`tabler:webhook` `iconify tabler:webhook` `iconify-icon tabler:webhook` `webhook` `iconify webhook`等格式，均无法正常显示。

## 配置
<details>
<summary>展开查看</summary>

```typescript
import { hopeTheme } from 'vuepress-theme-hope';
import sidebar from './sidebar';

export default hopeTheme({
    darkmode: 'switch',

    iconAssets: ["iconify", "fontawesome-with-brands"],
    iconPrefix: "",

    plugins: {
        prismjs: false,
        mdEnhance: {
            tabs: true,
            mermaid: true,
            tasklist: true,
            card: true,

            figure: true, // 图片描述
            imgLazyload: true,
            imgMark: true, // 夜间模式切换
            imgSize: true, // Resize

            container: true,

            align: true,
        },
    },
    navbar: ["/", ...],
    navbarLayout: {
        start: ["Brand"],
        end: ["Links", "Repo", "Search"],
    },

    sidebar: sidebar,

    repo: "...",

    footer: "MIT Licensed | Built on Vuepress Theme Hope",
    copyright: false,
})
```

</details>

## 复现方式
干净部署文档，使用对应主题配置，尝试使用iconify icon

## 相关信息
Vuepress: 2.0.0-beta.62
Vurepress-theme-hope: 2.0.0-beta.210
使用yarn和vite部署开发模式，未测试build

**Comment by Mister-Hope** [comment]
https://vuejs.press is using iconify.

Check your own configuration

**Reply by FYWinds** [reply]
I made a test.

> pnpm create vuepress-theme-hope add doc_test

First generate a new demo using the command from documentation.

Editing only this line of config
```ts
iconAssets: "iconfont",
```
to
```ts
iconAssets: "iconify",
```

The icon from iconify will display properly right now. Being transformed to `<iconify-icon>` in the html.
```
icon: tabler:webhook
```
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/74457127/6c9a859a-3e3f-4a1c-b6b1-dae98cb0cd18)

However, if change it to
```ts
iconAssets: ["iconify"],
```
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/74457127/af2fbe46-539f-49bd-ab82-42e45b45a4ee)
The icon won't display properly anymore, and the icon would be transformed to `<span class="">`.

Is that we cannot use multiple icon source together, or is it a bug?

**Reply by Mister-Hope** [reply, marked as answer]
iconify already supports mutiple icons sources so you probably do not need other icons. If you need a logo or something like that, you can just set the svg link there.

Also it use custom element, so the icon name are not element class name. How do I know a name is valid in iconify or not?

**Reply by FYWinds** [reply]
Yes, iconify itself is powerful enough and I have now changed other icons to iconify ones. Thanks for your answer!
